src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -150,6 +150,11 @@ bool YARPRobotStatePublisherModule::configure(ResourceFinder &rf)
         return false;
     }
 
+    // Set reduced model option
+    // The default value is false and considers streaming TFs of all the frames in the model
+    // If set to true, only the TFs of the links are streamed to transform server
+    this->reducedModelOption=rf.check("reduced-model",Value(0)).asBool();
```

Comment:
```suggestion
    this->reducedModelOption=rf.check("reduced-model", Value(0)).asBool();
```

Reply:
By the way, I think it is more usable to have boolean flags just as if `--reduced-model` is present enable the option, otherwise disable it.

Reply:
@traversaro I am not following the comment correctly

Reply:
Optional boolean arguments are tipically passed in a form of a flag that can be either be present or not, i.e. `--option`, not `--option true`. See for example the `ls` command in *nix systems: 
~~~
straversaro@iiticublap103:~/src/robotology-superbuild/robotology/robots-configuration/testSkin$ ls --help
Usage: ls [OPTION]... [FILE]...
List information about the FILEs (the current directory by default).
Sort entries alphabetically if none of -cftuvSUX nor --sort is specified.

Mandatory arguments to long options are mandatory for short options too.
  -a, --all                  do not ignore entries starting with .
  -A, --almost-all           do not list implied . and ..
      --author               with -l, print the author of each file
  -b, --escape               print C-style escapes for nongraphic characters
      --block-size=SIZE      scale sizes by SIZE before printing them; e.g.,
                               '--block-size=M' prints sizes in units of
                               1,048,576 bytes; see SIZE format below
  -B, --ignore-backups       do not list implied entries ending with ~
  -c                         with -lt: sort by, and show, ctime (time of last
                               modification of file status information);
                               with -l: show ctime and sort by name;
                               otherwise: sort by ctime, newest first
~~~
The options are `--all`, `--escape`, not `--all true`, `--escape`, etc etc.

Reply:
I see what you mean, if I understand well this boolean option should not be given as `--reduced-model true` but be simply used as the argument `--reduced-model` and is passed this option should be used. :+1:

Reply:
Exactly!

Reply:
Updated https://github.com/robotology/idyntree/pull/548/commits/9bc68bba7601c66be48c7db0bb6381805b897619